src/concurrency.md

```diff
@@ -72,6 +73,46 @@ fn main() {
 }
 ```
 
+[ex-rayon-parallel-search]: #ex-rayon-parallel-search
+<a name="ex-rayon-parallel-search"></a>
+## Search unordered items in parallel
```

Comment:
Hmm we might want to change the example title to something along the lines of
"Search items with given predicate in parallel". Originally the "unordered" part was to emphasize that we are not doing binary search but I think that mentioning the arbitrary predicate is a stronger usecase.